File: java/Problem151_ReverseWordsInString.java
// https://leetcode.com/problems/reverse-words-in-a-string/

class Solution {
    public String reverseWords(String s) {
        s = s.trim();
        StringBuilder sb = new StringBuilder();
        StringBuilder result = new StringBuilder();
        for(char c: s.toCharArray()){
            if(Character.isLetterOrDigit(c)){
                sb.append(c);
            }else if(sb.length()!=0){
                result.insert(0, sb);
                result.insert(0, " ");
                sb.setLength(0);
            }
        }
        result.insert(0, sb);
        return result.toString();
    }
}

File: go/problem151_reverse_words_in_string.go
// https://leetcode.com/problems/reverse-words-in-a-string/

func reverseWords(s string) string {
    // Fields is a split that use space as default
    fields := strings.Fields(s)

    // double for, firt create variable after receive values
    for i,j := 0, len(fields)-1; i<j; i,j = i+1, j-1 {
        fields[i], fields[j] = fields[j], fields[i]
    }

    return strings.Join(fields, " ")
}
